src/StartServerPlugin.js

```diff
@@ -56,20 +81,25 @@ export default class StartServerPlugin {
 
   _getScript(compilation) {
     const {entryName} = this.options;
-    const map = compilation.entrypoints;
-    const entry = map.get ? map.get(entryName) : map[entryName];
+    const entrypoints = compilation.entrypoints;
+    const entry = entrypoints.get
+      ? entrypoints.get(entryName)
+      : entrypoints[entryName];
     if (!entry) {
-      console.log(compilation);
+      this._info('compilation: %O', compilation);
       throw new Error(
-        `Requested entry "${entryName}" does not exist, try one of: ${(map.keys
-          ? map.keys()
-          : Object.keys(map)
+        `Requested entry "${entryName}" does not exist, try one of: ${(entrypoints.keys
+          ? entrypoints.keys()
+          : Object.keys(entrypoints)
         ).join(' ')}`
       );
     }
-    const entryScript = entry.chunks[0].files[0];
+
+    const entryScript = webpack.EntryPlugin
+      ? entry.runtimeChunk.files.values().next().value
+      : entry.chunks[0].files[0];
```

Comment:
This was erroring out for us, our `entry` looks like:

```{
  groupDebugId: 5000,
  options: { name: 'main' },
  _children: SortableSet(0) [Set] {
    _sortFn: [Function: sortById],
    _lastActiveSortFn: Symbol(not sorted),
    _cache: undefined,
    _cacheOrderIndependent: undefined
  },
  _parents: SortableSet(0) [Set] {
    _sortFn: [Function: sortById],
    _lastActiveSortFn: Symbol(not sorted),
    _cache: undefined,
    _cacheOrderIndependent: undefined
  },
  _blocks: SortableSet(0) [Set] {
    _sortFn: undefined,
    _lastActiveSortFn: Symbol(not sorted),
    _cache: undefined,
    _cacheOrderIndependent: undefined
  },
  chunks: [
    Chunk {
      id: 'main',
      ids: [Array],
      debugId: 1000,
      name: 'main',
      idNameHints: [SortableSet [Set]],
      preventIntegration: false,
      filenameTemplate: undefined,
      _groups: [SortableSet [Set]],
      runtime: undefined,
      files: [SetDeprecatedArray [Set]],
      auxiliaryFiles: [Set],
      rendered: true,
      hash: '520f6ac56d030c8035af02bba7724e96',
      contentHash: [Object: null prototype],
      renderedHash: '520f6ac56d030c8035af',
      chunkReason: undefined,
      extraAsync: false
    }
  ],
  origins: [
    {
      module: null,
      loc: [Object],
      request: '!!~/packages/razzle-dev-utils/monitor-loader.js!~/packages/razzle-dev-utils/monitor-loader.js'
    }
  ],
  _modulePreOrderIndices: Map(2) {
    NormalModule {
      dependencies: [Array],
      blocks: [],
      type: 'javascript/auto',
      context: '~/packages/razzle-dev-utils',
      needId: true,
      debugId: 29681,
      resolveOptions: {},
      factoryMeta: undefined,
      _warnings: undefined,
      _errors: undefined,
      buildMeta: {},
      buildInfo: [Object],
      presentationalDependencies: [Array],
      request: '~/packages/razzle-dev-utils/monitor-loader.js!~/packages/razzle-dev-utils/monitor-loader.js',
      userRequest: '~/packages/razzle-dev-utils/monitor-loader.js!~/packages/razzle-dev-utils/monitor-loader.js',
      rawRequest: '!!~/packages/razzle-dev-utils/monitor-loader.js!~/packages/razzle-dev-utils/monitor-loader.js',
      binary: false,
      parser: [JavascriptParser],
      generator: JavascriptGenerator {},
      resource: '~/packages/razzle-dev-utils/monitor-loader.js',
      matchResource: undefined,
      loaders: [Array],
      error: null,
      _source: [OriginalSource],
      _sourceSizes: undefined,
      _lastSuccessfulBuildMeta: {},
      _forceBuild: false,
      useSourceMap: true,
      _ast: null
    } => 0,
    ExternalModule {
      dependencies: [],
      blocks: [],
      type: 'javascript/dynamic',
      context: null,
      needId: true,
      debugId: 29683,
      resolveOptions: {},
      factoryMeta: undefined,
      _warnings: undefined,
      _errors: undefined,
      buildMeta: [Object],
      buildInfo: [Object],
      presentationalDependencies: undefined,
      request: 'webpack/hot/log-apply-result',
      externalType: 'commonjs',
      userRequest: 'webpack/hot/log-apply-result',
      useSourceMap: true
    } => 1
  },
  _modulePostOrderIndices: Map(2) {
    ExternalModule {
      dependencies: [],
      blocks: [],
      type: 'javascript/dynamic',
      context: null,
      needId: true,
      debugId: 29683,
      resolveOptions: {},
      factoryMeta: undefined,
      _warnings: undefined,
      _errors: undefined,
      buildMeta: [Object],
      buildInfo: [Object],
      presentationalDependencies: undefined,
      request: 'webpack/hot/log-apply-result',
      externalType: 'commonjs',
      userRequest: 'webpack/hot/log-apply-result',
      useSourceMap: true
    } => 0,
    NormalModule {
      dependencies: [Array],
      blocks: [],
      type: 'javascript/auto',
      context: '~/packages/razzle-dev-utils',
      needId: true,
      debugId: 29681,
      resolveOptions: {},
      factoryMeta: undefined,
      _warnings: undefined,
      _errors: undefined,
      buildMeta: {},
      buildInfo: [Object],
      presentationalDependencies: [Array],
      request: '~/packages/razzle-dev-utils/monitor-loader.js!~/packages/razzle-dev-utils/monitor-loader.js',
      userRequest: '~/packages/razzle-dev-utils/monitor-loader.js!~/packages/razzle-dev-utils/monitor-loader.js',
      rawRequest: '!!~/packages/razzle-dev-utils/monitor-loader.js!~/packages/razzle-dev-utils/monitor-loader.js',
      binary: false,
      parser: [JavascriptParser],
      generator: JavascriptGenerator {},
      resource: '~/packages/razzle-dev-utils/monitor-loader.js',
      matchResource: undefined,
      loaders: [Array],
      error: null,
      _source: [OriginalSource],
      _sourceSizes: undefined,
      _lastSuccessfulBuildMeta: {},
      _forceBuild: false,
      useSourceMap: true,
      _ast: null
    } => 1
  },
  index: 0,
  _runtimeChunk: Chunk {
    id: 'main',
    ids: [ 'main' ],
    debugId: 1000,
    name: 'main',
    idNameHints: SortableSet(0) [Set] {
      _sortFn: undefined,
      _lastActiveSortFn: Symbol(not sorted),
      _cache: undefined,
      _cacheOrderIndependent: undefined
    },
    preventIntegration: false,
    filenameTemplate: undefined,
    _groups: SortableSet(1) [Set] {
      [Circular *1],
      _sortFn: [Function: compareChunkGroupsByIndex],
      _lastActiveSortFn: Symbol(not sorted),
      _cache: undefined,
      _cacheOrderIndependent: undefined
    },
    runtime: undefined,
    files: SetDeprecatedArray(1) [Set] { 'main.js' },
    auxiliaryFiles: Set(1) { 'main.js.map' },
    rendered: true,
    hash: '520f6ac56d030c8035af02bba7724e96',
    contentHash: [Object: null prototype] { javascript: '95ee23a2924a50918a62' },
    renderedHash: '520f6ac56d030c8035af',
    chunkReason: undefined,
    extraAsync: false
  },
  _entrypointChunk: Chunk {
    id: 'main',
    ids: [ 'main' ],
    debugId: 1000,
    name: 'main',
    idNameHints: SortableSet(0) [Set] {
      _sortFn: undefined,
      _lastActiveSortFn: Symbol(not sorted),
      _cache: undefined,
      _cacheOrderIndependent: undefined
    },
    preventIntegration: false,
    filenameTemplate: undefined,
    _groups: SortableSet(1) [Set] {
      [Circular *1],
      _sortFn: [Function: compareChunkGroupsByIndex],
      _lastActiveSortFn: Symbol(not sorted),
      _cache: undefined,
      _cacheOrderIndependent: undefined
    },
    runtime: undefined,
    files: SetDeprecatedArray(1) [Set] { 'main.js' },
    auxiliaryFiles: Set(1) { 'main.js.map' },
    rendered: true,
    hash: '520f6ac56d030c8035af02bba7724e96',
    contentHash: [Object: null prototype] { javascript: '95ee23a2924a50918a62' },
    renderedHash: '520f6ac56d030c8035af',
    chunkReason: undefined,
    extraAsync: false
  }
}
```

i got it to work with the following code (which is also more defensive)

```ts
const runtimeChunk = webpack.EntryPlugin && (entry.runtimeChunk || entry._entrypointChunk)
    const runtimeChunkFiles = runtimeChunk && runtimeChunk.files && runtimeChunk.files.values()
    const entryScript = (runtimeChunkFiles && runtimeChunkFiles.next().value) || ((entry.chunks[0] || {}).files || [])[0]
    if (!entryScript) {
      this._error("Entry chunk not outputted: %O", entry)
      return
    }
```

Reply:
Thanks! What webpack version is this?

Reply:
5.0.0-beta.29

Reply:
Looks good to me 😀

Reply:
@fivethreeo/start-server-webpack-plugin@3.0.0-rc5 out now